composer/lib/dependabot/composer/file_parser.rb

```diff
@@ -166,7 +166,9 @@ def git_dependency_details(package_details, requirement)
 
       def lockfile_details(name:, type:)
         key = lockfile_key(type)
-        parsed_lockfile.fetch(key, []).find { |d| d["name"] == name }
+
+        fetchresults = parsed_lockfile.fetch(key, []) unless parsed_lockfile.nil?
+        fetchresults.find { |d| d["name"] == name } unless fetchresults.nil? || fetchresults.empty?
```

Comment:
The second argument to `Hash#fetch` will be returned if key is not in hash, but you're right that if the key is present but the value is nil, the result will be nil.

The `fetchresults.empty?` check is unnecessary because calling `#find` on an empty array is fine and will just return nil.

Here's another version to consider. Here I'm using the [safe navigation operator](https://ruby-doc.org/core-3.0.2/doc/syntax/calling_methods_rdoc.html#label-Safe+Navigation+Operator) to chain method calls that might return nil.

```ruby
      def lockfile_details(name:, type:)
        return unless parsed_lockfile

        key = lockfile_key(type)
        parsed_lockfile.fetch(key, [])&.find { |d| d["name"] == name }
      end
```

Reply:
I wanted to err on the side of readability and while the empty array should return nil, I just don't trust running find on an empty array.  I know it likely is not a problem in Ruby, and probably won't be in the foreseeable future, but I have just been caught flatfooted by changes in languages that I thought would never happen.  This way, we are sure we won't get that error if someone changes the way find is implemented in the future.  I can simplify as you put it, however, since it is cleaner and does the same.

The scenario was where the lock file had a line that said "packages-dev:null".  That threw an NPE because while the key was there, the value was null.